src/test/java/io/redis/examples/async/StringExample.java

```diff
@@ -0,0 +1,127 @@
+// EXAMPLE: set_tutorial
+package io.redis.examples.async;
+
+import io.lettuce.core.*;
+import io.lettuce.core.api.async.RedisAsyncCommands;
+import io.lettuce.core.api.StatefulRedisConnection;
+
+// REMOVE_START
+import org.junit.jupiter.api.Test;
+// REMOVE_END
+
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+// REMOVE_START
+import static org.assertj.core.api.Assertions.assertThat;
+// REMOVE_END
+
+public class StringExample {
+
+    // REMOVE_START
+    @Test
+    // REMOVE_END
+    public void run() {
+        RedisClient redisClient = RedisClient.create("redis://localhost:6379");
+
+        try (StatefulRedisConnection<String, String> connection = redisClient.connect()) {
+            RedisAsyncCommands<String, String> asyncCommands = connection.async();
+
+            // STEP_START set_get
+            CompletableFuture<Void> setAndGet = asyncCommands.set("bike:1", "Deimos").thenCompose(v -> {
+                System.out.println(v); // OK
+                // REMOVE_START
+                assertThat(v).isEqualTo("OK");
+                // REMOVE_END
+                return asyncCommands.get("bike:1");
+            })
+                    // REMOVE_START
+                    .thenApply(res -> {
+                        assertThat(res).isEqualTo("Deimos");
+                        return res;
+                    })
+                    // REMOVE_END
+                    .thenAccept(System.out::println) // Deimos
+                    .toCompletableFuture();
+            // STEP_END
+
+            // STEP_START setnx_xx
+            CompletableFuture<Void> setnx = asyncCommands.setnx("bike:1", "bike").thenCompose(v -> {
+                System.out.println(v); // false (because key already exists)
+                // REMOVE_START
+                assertThat(v).isFalse();
+                // REMOVE_END
+                return asyncCommands.get("bike:1");
+            })
+                    // REMOVE_START
+                    .thenApply(res -> {
+                        assertThat(res).isEqualTo("Deimos");
+                        return res;
+                    })
+                    // REMOVE_END
+                    .thenAccept(System.out::println) // Deimos (value is unchanged)
+                    .toCompletableFuture();
+
+            // set the value to "bike" if it already exists
+            CompletableFuture<Void> setxx = asyncCommands.set("bike:1", "bike", SetArgs.Builder.xx())
+                    // REMOVE_START
+                    .thenApply(res -> {
+                        assertThat(res).isEqualTo("OK");
+                        return res;
+                    })
+                    // REMOVE_END
+                    .thenAccept(System.out::println) // OK
+                    .toCompletableFuture();
```

Comment:
This is not necessarily bad, but you are not calling any of the methods that would guarantee that the main thread would not end before the executor threads. 

As far as my understanding of how CompletableFuture works the code inside the stages / futures is executed by some thread pool (if no executor is provided then that would be the ForkJoinPool#commonPool) and without calling  get() or join() for example *theoretically* the main thread might end before the thread that executes the future does. IMHO it would be good to implement some sort of synchronization here, simplest would be to call get(), but you can also do some sort of awat-all for all the futures in the end of the main method